crates/torii/core/src/executor.rs

```diff
@@ -49,30 +45,36 @@ pub enum QueryType {
     EventMessage(Ty),
     RegisterModel,
     StoreEvent,
+    Execute,
     Other,
 }
 
-impl QueryQueue {
-    pub fn new(pool: Pool<Sqlite>) -> Self {
-        QueryQueue { pool, queue: VecDeque::new() }
-    }
+pub struct Executor<'c> {
+    pool: Pool<Sqlite>,
+    transaction: Transaction<'c, Sqlite>,
+    publish_queue: VecDeque<BrokerMessage>,
+    rx: UnboundedReceiver<QueryMessage>,
+}
 
-    pub fn enqueue<S: Into<String>>(
-        &mut self,
-        statement: S,
-        arguments: Vec<Argument>,
-        query_type: QueryType,
-    ) {
-        self.queue.push_back((statement.into(), arguments, query_type));
-    }
+pub struct QueryMessage {
+    pub statement: String,
+    pub arguments: Vec<Argument>,
+    pub query_type: QueryType,
+}
+
+impl<'c> Executor<'c> {
+    pub async fn new(pool: Pool<Sqlite>) -> Result<(Self, UnboundedSender<QueryMessage>)> {
```

Comment:
Let's monitor the `unbounded` behavior in this case as a lot of messages are expected. To see if a buffer is preferable, but we may have some work to make everything async as you mentioned.